### PR TITLE
Allow cubeviz image viewer to display coords for 2D data

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request is to address the cubeviz coords display failing for 2D data in the image viewer.

This is a direct follow-up of #1315

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)? N/A
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/cubeviz/plugins/viewers.py [hunk 1]
```diff
@@ -91,13 +91,16 @@ def on_mouse_or_key_event(self, data):
...
+            if (-0.5 < x < image.shape[0] - 0.5 and -0.5 < y < image.shape[1] - 0.5
                     and hasattr(visible_layers[0], 'attribute')):
                 attribute = visible_layers[0].attribute
-                value = image.get_data(attribute)[int(round(x)), int(round(y)),
-                                                  self.state.slices[-1]]
+                if len(image.shape) == 3:
+                    value = image.get_data(attribute)[int(round(x)), int(round(y)),
+                                                      self.state.slices[-1]]
+                elif len(image.shape) == 2:
+                    value = image.get_data(attribute)[int(round(x)), int(round(y))]
```
[the remark this answers is not in the document]
Comment (javerbukh) [reply jdaviz/configs/cubeviz/plugins/viewers.py:102]: I am not sure! How do I check that?

[the remark this answers is not in the document]
Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/viewers.py:102]: I usually do that with asymmetric dataset with known values (e.g., `np.arange()`). Then mouseover the corners and make sure X, Y, and value are what you expect them to be. It is easier to tell if data is not symmetric.

[the remark this answers is not in the document]
Comment (javerbukh) [reply jdaviz/configs/cubeviz/plugins/viewers.py:102]: I'm a little confused. I tried the following code snippet and the data presented in the viewer does not line up with the value I see at the top. If I switch the x and y axes, it fails altogether if I move the cursor to the edges of the data. Any advice?

```
from astropy.nddata import CCDData
from matplotlib import pyplot as plt   # doctest: +SKIP
x = CCDData([[1,2,3], [4,5,6]], unit='adu')

viz.app.data_collection["Test"] = x
```

[the remark this answers is not in the document]
Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/viewers.py:102]: I dunno... Is this another manisfestation of #1323 ? Maybe we need to fix the underlying linking problem before coming back to this. For now, how about just set `value` to empty string if it is 2D?

**Review by pllim — APPROVED** [review]
Actually, I think this might be correct, at least X and Y seems sensible when I tested with the data from here:

https://github.com/spacetelescope/jdaviz/blob/3516b14f2d21d985490a01ac537e8a060aeaa8f6/jdaviz/configs/cubeviz/plugins/tests/test_parsers.py#L10

If we see weird behavior due to WCS swapping in some other data, we can deal with it later. Thanks for fixing my bug!

**Pull request merged**